anta/tests/interfaces.py

```diff
@@ -438,3 +483,78 @@ def test(self) -> None:
             self.result.is_failure(f"Some L2 interfaces do not have correct MTU configured:\n{wrong_l2mtu_intf}")
         else:
             self.result.is_success()
+
+
+class VerifyInterfacesSpeed(AntaTest):
+    """
+    Verifies the speed, lanes, auto-negotiation status, and mode as full duplex for interfaces.
+    If speed is auto then verify auto-negotiation as success and mode as full duplex.
+    If speed is auto with a value(auto 10g) then verify auto-negotiation as success, mode as full duplex and speed/lanes as per input.
+    If speed is forced with a value(forces 10g) then verify mode as full duplex and speed as per input.
+    If speed with lane(100g-8) then verify mode as full duplex and speed/lanes as per input.
+
+    Expected Results:
+        * Success: The test will pass if an interface is configured with the correct speed, lanes, auto-negotiation and mode as full duplex.
+        * Failure: The test will fail if an interface is not found, the speed or alnes does not match with input,
+                   auto-negotiation is not correct or mode is not full duplex.
+    """
+
+    name = "VerifyInterfacesSpeed"
+    description = "Verifies the speed, lanes, auto-negotiation status, and mode as full duplex for interfaces."
+    categories = ["interfaces"]
+    commands = [AntaCommand(command="show interfaces")]
+
+    class Input(AntaTest.Input):
+        """Inputs for the VerifyInterfacesSpeed test."""
+
+        interfaces: List[Interfaces]
+        """List of interfaces to be tested"""
+
+        class Interfaces(BaseModel):
+            """Detail of an interface"""
+
+            interface: SpeedInterface
+            """Name of the interface"""
+            speed: str = Field(..., pattern=SpeedPattern)
+            """Speed of an interface in Gigabits per second"""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        self.result.is_success()
+        command_output = self.instance_commands[0].json_output
+
+        # Iterate over all the interfaces
+        for interface in self.inputs.interfaces:
+            intf = interface.interface
+
+            # Check if interface exists
+            if not (interface_output := get_value(command_output, f"interfaces.{intf}")):
+                self.result.is_failure(f"Interface `{intf}` not found.")
+                continue
+
+            auto_negotiation = interface_output.get("autoNegotiate")
+            duplex = interface_output.get("duplex")
+            actual_speed = interface_output.get("bandwidth")
+            actual_lanes = interface_output.get("lanes")
+            speed, lanes = extract_speed_and_lane(interface.speed)
+
+            # Collecting actual interface details
+            actual_interface_output = {
+                "auto negotiation": auto_negotiation if "auto" in interface.speed else "None",
```

Comment:
"None" need to replace with Null

Reply:
Why are you using "None"? It looks wrong no?

Reply:
To create a generalize dict of actual and expected speed details of various type of speeds i used "None" in case that parameter is not needed for verification. For example if speed is 100g-4 in that case we dont need to validate auto negotiation hence updating auto negotiation key as None in both actual and expected output. So while comparing both actual and expected parameters match. I know if speed is 100g-4 in that case we should not define auto negotiation in actual and expected output but to achieve this we will have multiple condition for each type.

Let me know in case we need to change it.